modules/proxy/proxy_util.c

```diff
@@ -3544,12 +3823,10 @@ static int proxy_connection_create(const char *proxy_function,
          * the peer reset the connection already; ap_create_connection()
          * closed the socket
          */
-        ap_log_error(APLOG_MARK, APLOG_DEBUG, 0,
+        ap_log_error(APLOG_MARK, APLOG_ERR, 0,
                      s, APLOGNO(00960) "%s: an error occurred creating a "
                      "new connection to %pI (%s)", proxy_function,
                      backend_addr, conn->hostname);
-        /* XXX: Will be closed when proxy_conn is closed */
-        socket_cleanup(conn);
```

Comment:
Every caller of ap_proxy_connection_create[_ex]() should release/invalidate the connection on failure already, like with any other ap_proxy_ function taking a proxy_conn_rec which fails?

Reply:
Good point, but in order to be save shouldn't we do at least a `conn->close = 1` here in case the caller only releases the connection but forgets to update the `close` flag before? But I agree that the current consumers in our code do it properly.